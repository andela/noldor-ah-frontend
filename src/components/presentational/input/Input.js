import React from 'react';
import propTypes from 'prop-types';
import '../styles/Input.scss';

const Input = (props) => {
  const {
    name, title, type,
    value, onChange, placeholder,
    className, required, pattern,
    id, parentClass,
  } = props;

  return (
    <div className={parentClass}>
      <label htmlFor={name} className="form-label">
        {title}
      </label>
      <input
        id={id}
        className={className}
        name={name}
        type={type}
        value={value}
        onChange={onChange}
        placeholder={placeholder}
        required={required}
        pattern={pattern}
      />
    </div>
  );
};

Input.propTypes = {
  name: propTypes.string,
  title: propTypes.string,
  type: propTypes.string,
  value: propTypes.string,
  onChange: propTypes.func,
  placeholder: propTypes.string,
  className: propTypes.string,
  parentClass: propTypes.string,
  id: propTypes.string,
  required: propTypes.bool,
  pattern: propTypes.string,
};

Input.defaultProps = {
  name: '',
  title: '',
  type: '',
  value: '',
  placeholder: '',
  className: 'form-control',
  id: '',
  required: false,
  pattern: null,
  parentClass: 'form-group'
};

export default Input;
