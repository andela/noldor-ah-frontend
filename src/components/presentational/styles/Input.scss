$ah-grey: #9D9D9D;
$ah-primary:  #E66869;

.form-control {
  border-style: none;
  width: 100%;
  padding: .5em 0;
  border-bottom: 2px solid $ah-grey;
  font-size: 1rem;
}

.form-control:hover, .form-control:focus {
  border-bottom: 2px ridge $ah-primary;
  outline: 0;
}

.form-control ::placeholder {
  font-weight: thin;
  color: $ah-grey;
}