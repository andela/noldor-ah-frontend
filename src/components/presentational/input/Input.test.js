import React from 'react';
import { shallow } from 'enzyme';
import Input from './Input';

describe('<Input /> shallow rendering tests', () => {
  const handleChange = jest.fn();
  const wrapper = shallow(
    <Input
      id="mia"
      className="test"
      type="text"
      value="Smurf"
      name="Mia"
      title="Mia"
      onChange={handleChange}
    />
  );

  it('should render Input with props successfully', () => {
    expect(wrapper.find('input').hasClass('test')).toBe(true);
    expect(wrapper.find('#mia').exists()).toBe(true);
    expect(wrapper.find('input[type="text"]').exists()).toBe(true);
    expect(wrapper.find('label').exists()).toBe(true);
    expect(wrapper.find('label').text()).toBe('Mia');
  });

  it('should pass in the onChange prop successfully', () => {
    wrapper.find('input').simulate(
      'change',
      {
        target: {
          name: 'Mia',
          value: 'This is me'
        }
      }
    );
    expect(handleChange.mock.calls.length).toBe(1);
  });

  it('matches the snapshot', () => {
    const tree = shallow(<Input />);
    expect(tree).toMatchSnapshot();
  });
});
