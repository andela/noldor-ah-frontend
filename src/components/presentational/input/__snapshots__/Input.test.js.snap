// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> shallow rendering tests matches the snapshot 1`] = `
<div
  className="form-group"
>
  <label
    className="form-label"
    htmlFor=""
  />
  <input
    className="form-control"
    id=""
    name=""
    pattern={null}
    placeholder=""
    required={false}
    type=""
    value=""
  />
</div>
`;
